### PR TITLE
Added extra spare pin to P1 (P1S6) with GPIO and PWM support.

### PR DESCRIPTION
**Pull request by technobly**
Added extra spare pin to P1 (P1S6) with GPIO and PWM support.  Implements #1059 

>This is the TESTMODE pin 33 on the P1 module, and to use it properly the Wi-Fi Powersave Clock must be disabled.  See docs below.

---

Doneness:

- [x] Contributor has signed CLA
- [x] Problem and Solution clearly stated
- [ ] Code peer reviewed
- [x] API tests compiled
- [x] Run unit/integration/application tests on device
- [ ] Add documentation (for 0.7.0-rc.1)

https://github.com/spark/docs/commit/c63c4f0e03b6fce0361aa221ac8c6ddb329ab19a

- [ ] Add to CHANGELOG.md after merging (add links to docs and issues)

### FEATURES

- [P1] Added extra spare pin to P1 (P1S6) with GPIO and PWM support.  Implements [#1059](https://github.com/spark/firmware/issues/1059)

### CODE REVIEW

File: hal/src/stm32f2xx/core_hal_stm32f2xx.c [hunk 1]
```diff
@@ -314,6 +319,11 @@ void HAL_Core_Config(void)
 #if PLATFORM_ID==8 // Additional pins for P1
     for (pin_t pin=24; pin<=29; pin++)
         HAL_Pin_Mode(pin, INPUT);
+    const uint8_t* data = (const uint8_t*)dct_read_app_data(DCT_RADIO_FLAGS_OFFSET);
+    uint8_t current = (*data);
+    if ((current&3) == 0x2) {
```
Comment (m-mcgowan) [inline hal/src/stm32f2xx/core_hal_stm32f2xx.c]: I know this is my (shoddy!) code, and felt the same way when I wrote it....it's probably best to pull these magic numbers into a function like

```
bool isWiFiPowersaveClockDisabled() {
 const uint8_t* data = (const uint8_t*)dct_read_app_data(DCT_RADIO_FLAGS_OFFSET);
  uint8_t current = (*data);
  return ((current&3) == 0x2);
}
```

File: user/tests/wiring/no_fixture/pwm.cpp [hunk 2]
```diff
@@ -29,15 +29,26 @@
 
 uint8_t pwm_pins[] = {
 
-#if defined(STM32F2XX)
+#if (PLATFORM_ID == 0) // Core
```
Comment (m-mcgowan) [inline user/tests/wiring/no_fixture/pwm.cpp]: lovely! I'm surprised we didn't hit the need to specify the PWM pins per platform before :-)

Comment (technobly) [inline user/tests/wiring/no_fixture/pwm.cpp]: Yeah I don't know why, but we were not testing all PWM pins on all platforms before this change.  I was pleased to see it pass first time on all platforms though! :)

File: hal/src/stm32f2xx/core_hal_stm32f2xx.c (hunk 1 above)
Comment (technobly) [inline hal/src/stm32f2xx/core_hal_stm32f2xx.c]: See commit https://github.com/spark/firmware/pull/1120/commits/b3f8160de6428b597b2d35f576303dd46677abfc for resurrection of a fairy ;-)

**Comment by technobly** [comment]
Here's also the result of the power comparison when running the TEST=wiring/no_fixture test while measuring current on VUSB, with powersave clock enabled and disabled.  As seen there are no notable differences.
powersave

**Pull request merged**